TFT/src/User/API/menu.c

```diff
@@ -819,7 +819,16 @@ void menuDrawTitle(const uint8_t *content)
   GUI_SetBkColor(infoSettings.title_bg_color);
   if (content)
   {
-    GUI_DispLenString(10, start_y, content, LCD_WIDTH - 20, true);
+    if (MENU_IS(menuPrinting) && infoSettings.filename_extension == 0)
+    {
+      hideExtension((char *)content);
```

Comment:
function menuDrawTitle is a general function to display any kind of text in the title bar (not only a filename). E.g. it is called also by ListManager, ScreenSetting, Speed etc... None of those is passing a string with an additional extension flag at the end, so the restoreExtension will provide an out of buffer read in those use cases

Reply:
You missed the point that `hideExtension((char *)content)` and `restoreExtension((char *)content)` is called only if the conditional ` if (MENU_IS(menuPrinting) && infoSettings.filename_extension == 0)` is true. Can you guess where does "content" points in that case?

Reply:
You missed both my previous explanations. Even the Printing menu can pass a "standard" (no extension flag) string. And you also made a change in your PR where this happens. I suggested you to apply the hideExtension function directly in Printing menu (in the only point where you need it). It will be completely safe and logical and the code is even reduced in size (no need for extra checks, no need to invoke restoreExtension).
Sorry, but I cannot spend other time in such kind of infinite discussions

Reply:
> Even the Printing menu can pass a "standard" (no extension flag) string. And you also made a change in your PR where this happens.

The only time menuDrawTitle() is called in printing menu is when menuDrawPage() is called in the printing menu. There menuDrawPage() is impossible to pass a pointer to "standard" (no extension flag) char array because "printingItems.title.address" will always point to "infoFile.longFile[infoFile.fileIndex]" or "infoFile.file[infoFile.fileIndex]". Both of them already are extension flagged strings.

> I suggested you to apply the hideExtension function directly in Printing menu (in the only point where you need it). It will be completely safe and logical and the code is even reduced in size (no need for extra checks, no need to invoke restoreExtension).

You might have suggested it but it  implies further complications. If I only apply the hideExtension function than infoPrintSummary will  have the filename without extension.  To avoid that than extension must be restored after menuDrawPage() is called. This has another drawback, when menuReDrawCurTitle() is called than it would display again the file with extension. 

> I cannot spend other time in such kind of infinite discussions

No need to. It seems all your valid worries and usable suggestions are settled now.